### DISCUSSION TITLE
Debugging Python code that calls C++ code

### DISCUSSION THREAD

**Discussion by mvadari** [pybind/pybind11#4531] [Q&A]
I'm trying to write a Python module that calls C++ code (with a pybind11 wrapper). That Python code is then embedded in a C++ codebase. The idea is that you can write Python to modify the state of a C++ project. I'm getting an error while trying to debug (the Python code is erroring when I call a wrapped C++ function) and I'm not really sure the best way to go about it. Here's the stack trace:
```
* thread #5, name = 'doJob: RPC-Client', stop reason = EXC_BAD_ACCESS (code=1, address=0x28)
  * frame #0: 0x0000000129cb9d71 libpython3.11.dylib`new_threadstate + 17
    frame #1: 0x0000000129cbb1aa libpython3.11.dylib`PyGILState_Ensure + 74
    frame #2: 0x0000000100a10a49 rippled`pybind11::detail::get_internals(this=0x000000012faf9250)::gil_scoped_acquire_local::gil_scoped_acquire_local() at internals.h:426:44
    frame #3: 0x0000000100a0f5b5 rippled`pybind11::detail::get_internals(this=0x000000012faf9250)::gil_scoped_acquire_local::gil_scoped_acquire_local() at internals.h:426:65
    frame #4: 0x0000000100a0ee1d rippled`pybind11::detail::get_internals() at internals.h:431:7
    frame #5: 0x0000000100a0ecd8 rippled`pybind11::gil_scoped_acquire::gil_scoped_acquire(this=0x000000012faf9378) at gil.h:56:27
    frame #6: 0x0000000100a0d5c5 rippled`pybind11::gil_scoped_acquire::gil_scoped_acquire(this=0x000000012faf9378) at gil.h:55:44
    frame #7: 0x0000000100a0d50d rippled`pybind11::error_already_set::what(this=0x000060000331d0c0) const at pybind11.h:2630:24
```

Would appreciate any help. I don't know how to get the original C++ error.

**Comment by Dariusz1989** [comment]
Looks like we may have a similar issue... https://github.com/pybind/pybind11/discussions/4477

I take you attaching ur IDE to c++ app and try to break point?

**Reply by mvadari** [reply]
Yes. I was actually able to figure out the issue (it was with my pybind11 code). A couple of try-except loops got me an error message.

**Reply by Dariusz1989** [reply]
So you can import debug build pyd file in to python ? Can you breakpoint using IDE & attaching process the c++ code?

**Reply by mvadari** [reply]
I couldn't actually get the debugger to work the way I wanted to, but I'm pretty sure I was also calling the breakpoint in the wrong section of code - I stopped digging into that as soon as I figured out how to fix my issue.

**Reply by Dariusz1989** [reply]
So when you build it, to you build it in debur or release mode? When ever I build it as Debug & try to import resulting pyd I get a crash :/

**Reply by mvadari** [reply]
I haven't been having any issues with that, sorry.
